scripts/export.py

```diff
@@ -0,0 +1,143 @@
+
+    conn = pyathena.connect(
+        s3_staging_dir=os.getenv(
+            "AWS_ATHENA_S3_STAGING_DIR", "s3://ccao-athena-results-us-east-1/"
+        ),
+        region_name=os.getenv("AWS_REGION", "us-east-1"),
+    )
+
+    FLAG_LATEST_VERSION_QUERY = """
+        SELECT meta_sale_document_num, MAX(version) AS version
+        FROM sale.flag
+        GROUP BY meta_sale_document_num
+    """
+
+    logger.info("Querying the count of active flags")
+
+    cursor = conn.cursor()
+    cursor.execute(f"SELECT COUNT(*) FROM ({FLAG_LATEST_VERSION_QUERY}) flags")
+    expected_num_flags = cursor.fetchall()[0][0]
+
+    logger.info(f"Got {expected_num_flags} active flags")
+
+    FLAG_QUERY = f"""
+    SELECT
+        sale.pin AS {PIN_FIELD},
+        sale.sale_key AS {SALE_KEY_FIELD},
+        CASE
+            WHEN flag.sv_is_outlier = TRUE
+            THEN 'Y'
+            ELSE 'N'
+        END AS {IS_OUTLIER_FIELD},
+        flag.sv_outlier_type AS {OUTLIER_TYPE_FIELD},
+        flag.run_id AS {RUN_ID_FIELD}
+    FROM sale.flag AS flag
+    -- Filter flags for the most recent version
+    INNER JOIN ({FLAG_LATEST_VERSION_QUERY}) AS flag_latest_version
+        ON flag.meta_sale_document_num = flag_latest_version.meta_sale_document_num
+        AND flag.version = flag_latest_version.version
+    INNER JOIN default.vw_pin_sale AS sale
+        ON flag.meta_sale_document_num = sale.doc_no
+    """
+
+    logger.info("Querying sales with flags")
+
+    cursor.execute(FLAG_QUERY)
+    flag_df = pyathena.pandas.util.as_pandas(cursor)
+
+    num_flags = len(flag_df.index)
+    logger.info(f"Got {num_flags} sales with flags")
+
+    logger.info("Transforming columns")
+
+    # Transform outlier type column from string to code
+    flag_df[OUTLIER_TYPE_FIELD] = flag_df[OUTLIER_TYPE_FIELD].replace(
+        OUTLIER_TYPE_CODES
+    )
+
+    logger.info("Running data integrity checks")
+
+    # Run some data integrity checks
+    not_null_fields = [PIN_FIELD, SALE_KEY_FIELD, RUN_ID_FIELD]
+    for field in not_null_fields:
+        assert flag_df[flag_df[field].isnull()].empty, f"{field} contains nulls"
+
+    assert flag_df[
+        ~flag_df[OUTLIER_TYPE_FIELD].isin(OUTLIER_TYPE_CODES.values())
+    ].empty, f"{OUTLIER_TYPE_FIELD} contains invalid codes"
+
+    assert flag_df[
+        (flag_df[IS_OUTLIER_FIELD] == "Y")
+        & (flag_df[OUTLIER_TYPE_FIELD] == OUTLIER_TYPE_CODES["Not outlier"])
+    ].empty, (
+        f"{OUTLIER_TYPE_FIELD} cannot be {OUTLIER_TYPE_CODES['Not outlier']} "
+        f"when {IS_OUTLIER_FIELD} is Y"
+    )
+
+    assert flag_df[
+        (flag_df[IS_OUTLIER_FIELD] == "N")
+        & (flag_df[OUTLIER_TYPE_FIELD] != OUTLIER_TYPE_CODES["Not outlier"])
+    ].empty, (
+        f"{OUTLIER_TYPE_FIELD} must be {OUTLIER_TYPE_CODES['Not outlier']} "
+        f"when {IS_OUTLIER_FIELD} is N"
+    )
+
+    assert (
+        num_flags == expected_num_flags
+    ), f"Expected {expected_num_flags} total sales, got {num_flags}"
```

Comment:
At some point it would be nice to copy these tests over to dbt to test the iasWorld flag table, but however we decide to do that, I still think it's useful to have some guardrails here that will warn us if we modify the flag query and accidentally introduce a bad join or select that clobbers the data in some unanticipated way.

Reply:
**praise:** Agree we should move it over, but this is great. Would like to see more defensive coding like this elsewhere in our repos.